packages/core/src/tracing/instrumentation/messaging/natsStreaming.js

```diff
@@ -194,6 +194,7 @@ function addErrorToSpan(err, span) {
   if (err) {
     span.ec = 1;
 
+    // TODO: special logic of appending error
```

Comment:
Agree. This should be handled as a separate PR after this merge for nats streaming as an action item of #2181

setErrorDetails needs to handle err being a String - some libs have very bad error handling